packages/core/src/storage.ts

```diff
@@ -208,6 +208,49 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  aggregateBy?: string
+  filter?: {
```

Comment:
Is it document attributes? How it will be applied, could it be 'or' or 'and'? Could it be from multiple classes? like Employee and Organization fields?

Reply:
Right now It's elastic document attributes not our Doc attribute and it only could be 'and'. So as we are dealing with elastic documents there is no knowledge about our classes and if the user want to make filters on class attributes he has to use docKey or something on his own. Right now this is about what we need for mentions now. 

And we could improve this in several ways, just let's discuss what do we want to have right now.

What could we improve:
1. Make more clear structure to be able to make more complex conditions. We even could make it mongo-like conditions. Or we could just add more like "contains" and "contains not". Do work on this I would need to make some decent tests for elastic because I cannot see scenarios for this in mentions right now.
2. We could introduce docKey like mapping inside elastic adapter. I'm not sure how deep we want to get the knowledge about our hierarchy there, but this is obviously possible solution.

Reply:
I'm not sure about usage scenario to pass _class filter here. Probable we need to have it in parameters. And we need to think about more examples to use filters to fully be sure about do we need it or not.

Reply:
But now search is available only for certain categories https://github.com/hcengineering/anticrm/blob/main/models/contact/src/index.ts#L711
We have ObjectSearchCategory in a model and it gives us necessary labels and icons. And in PR it gives also a component to render result. 

If we wouldn't filter by _class how can we ensure that we only asking for results that we need? Example: Backlinks. They are in index for some reason but we do not want to search for them. I could image there might be more object which shouldn't be available through the common search, or which shouldn't be mentioned. How to deal with it without filter by _class?